### ISSUE TITLE
MissingReturnType is not emitted for __invoke()

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#954]
https://getpsalm.org/r/1a564cafda
```php
<?php
  
class C {
  public function ordinary() {}
}
class D {
  public function __invoke() {}
}
```
**Expected**: MissingReturnType emitted for both methods
**Actual**:
```
Psalm output (using commit 91100ed): 

INFO: MissingReturnType - 4:3 - Method C::ordinary does not have a return type, expecting void
```

**Comment by weirdan** [comment]
Is it useful to assume that, though? I don't see why it would be.

**Comment by muglug** [comment]
I'm just not sure this is a valuable thing to enforce. Psalm is happy to check that the return types aren't correct, but I don't want to enforce it.

**Comment by weirdan** [comment]
Seems no less valuable than enforcing function return types, and Psalm does that. Defining `__invoke` makes objects of that class to behave as functions, so applying the same level of rigor seems natural.

**Comment by weirdan** [comment]
Still puzzled as to why it was rejected. Mixed, of course, is still reported at call site, but it's not where it needs to be fixed. `__get()` usually has multiple possible return types (e.g. one for every accepted property name), and specifying huge union may be unwieldy especially since you can document it with `@property`, but `__invoke()` is more likely to have narrower return type and has no alternative way of documenting it.

Reason for necroposting: got bitten by this again and was going to file a new issue.
